uiautomationutilities/perf_tool/readme.md

```diff
@@ -1,155 +1,157 @@
+	- `CodeMarkerManager.getInstance().markCode(CodeMarkersConstants.ACQUIRE_TOKEN_SILENT_START);`
+	- `CodeMarkerManager.getInstance().markCode(CodeMarkersConstants.ACQUIRE_TOKEN_SILENT_FUTURE_OBJECT_CREATION_END);`
+- Write a new test case that targets the flow where code markers have been added, see existing test case [TestCasePerf.java](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/perf/TestCasePerf.java) for non-brokered auth and [TestCasePerfBrokered.java](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/perf/TestCasePerfBrokered.java) for brokered auth.
+- In the new test case, before performing the particular scenario, add following code snippet
+	- Enable the Code marker: `CodeMarkerManager.getInstance().setEnableCodeMarker(true);`
+	- Setup the scenario code defined in step #1: `CodeMarkerManager.getInstance().setPrefixScenarioCode("100");`
+	- If we are having more than one iterations of same scenario, i.e. running same code multiple times, 
+		- Make sure to clear the previous run's markers in the start of the iteration as follows: `CodeMarkerManager.getInstance().clearMarkers();`
+		- Also write the content to the file using [FileAppender](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/appender/FileAppender.java) such as [here](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/perf/TestCasePerf.java#L119-L124)
+	- Also add followings after performing the particular scenario:
+		- Clear the codemarkers: `CodeMarkerManager.getInstance().clearMarkers();`
+		- Disable the codemarkers: `CodeMarkerManager.getInstance().setEnableCodeMarker(false);`
+- In the C# tool code, open the file [PerfDataConfiguration.xml](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml) and perform following actions:
+	- In tag `MeasurementsConfigurations`, add a new `MeasurementsConfiguration` with giving the startMarker and endMarker as given in the step 2. 
+	- In tag, `Scenarios`, add a new scenario with the Measurement ID of the MesurementConfiguration just added in previous sub-step.
+- Push the common code and msal native code to dev branch by raising PRs.
+- Push the C# changes of its repo (i.e. common) into dev branch by raising a PR.
+- Deploy C# tool into the pipeline. (Pipeline information given at the start of this doc). We can deploy C# tool even before merging of PR if need be. As this deployment is done after building at the local machine.
```

Comment:
So... adding new scenarios means the tool needs to be rebuilt and redeployed to the pipeline? Why? There's a plan to remove this step in the future?

Reply:
The PerfDataConfiguration.xml is the only file that needs to be updated - will look for a better way of updating the file without bundling the whole report tool